dashboard/src/reducers/datasetListReducer.js

```diff
@@ -1,33 +1,64 @@
-import {
-  GET_PUBLIC_DATASETS,
-  FAVORITED_DATASETS,
-  UPDATE_PUBLIC_DATASETS,
-} from "../actions/types";
+import * as CONSTANTS from "assets/constants/browsingPageConstants";
+import * as TYPES from "actions/types";
 
 const initialState = {
   publicData: [],
   favoriteRepoNames: [],
   tableData: [],
+  offset: 0,
+  limit: CONSTANTS.LIMIT,
+  perPage: CONSTANTS.DEFAULT_PER_PAGE,
+  totalDatasets: 0,
+  searchKey: "",
+  filter: {
+    startDate: "",
+    endDate: "",
+  },
 };
 
 const DatasetListReducer = (state = initialState, action = {}) => {
   const { type, payload } = action;
   switch (type) {
-    case GET_PUBLIC_DATASETS:
+    case TYPES.GET_PUBLIC_DATASETS:
       return {
         ...state,
         publicData: [...payload],
-        tableData: [...payload],
       };
-    case FAVORITED_DATASETS:
+    case TYPES.FAVORITED_DATASETS:
       return {
         ...state,
         favoriteRepoNames: [...payload],
       };
-    case UPDATE_PUBLIC_DATASETS:
```

Comment:
`UPDATE_PUBLIC_DATASETS` was a better name than `GET_PUBLIC_DATASETS` (it's weird to have a "get" verb in this context).

Reply:
Revised the name to be `UPDATE_PUBLIC_DATASETS`